docs/user-guide/gpus.md

```diff
@@ -0,0 +1,136 @@
+---
+assignees:
+- vishh
+title: GPU Support
+---
+
+Kubernetes includes **experimental** support for managing NVIDIA GPUs spread across nodes.
+This page describes how users can consume GPUs and the current limitations.
+
+## Pre-requisites
+
+1. Kubernetes nodes have to be pre-installed with Nvidia drivers. Kubelet will not detect Nvidia GPUs otherwise.
+2. A special **alpha** feature gate `Accelerators` has to be set to true across the system: `--feature-gates="Accelerators=true"`.
+3. Nodes must be using `docker engine` as the container runtime.
+
+The nodes will automatically discover and expose all Nvidia GPUs as a schedulable resource.
+
+## API
+
+Nvidia GPUs can be consumed via container level resource requirements using the resource name `alpha.kubernetes.io/nvidia-gpu`.
+
+```yaml
+apiVersion: v1
+kind: pod
+spec: 
+  containers: 
+    - 
+      name: gpu-container-1
+      resources: 
+        limits: 
+          alpha.kubernetes.io/nvidia-gpu: 2
+    - 
+      name: gpu-container-2
+      resources: 
+        limits: 
+          alpha.kubernetes.io/nvidia-gpu: 3
+```
+
+- GPUs can be specified in the `limits` section only.
+- Containers (and pods) do not share GPUs.
+- Each container can request one or more GPUs.
+- It is not possible to request a portion of a GPU.
+- Nodes are expected to be homogenous, i.e. run the same GPU hardware.
+
+If your nodes are running different versions of GPUs, then use Node Labels and Node Selectors to schedule pods to appropriate GPUs.
+Following is an illustration of this workflow:
+
+As part of your Node bootstrapping, identify the GPU hardware type on your nodes and expose it as a node label.
```

Comment:
The script I posted should ideally work with kubeadm/kops too. I haven't tested it yet. 

@mikedanese @justinsb Is `/etc/default/kubelet` a valid configuration medium in kubeadm and kops?